core/src/main/java/io/grpc/ProxyParameters.java

```diff
@@ -58,4 +78,41 @@ public boolean equals(Object o) {
   public int hashCode() {
     return Objects.hashCode(proxyAddress, username, password);
   }
+
+  /**
+   * Create a new builder.
+   */
+  public static Builder forAddress(InetSocketAddress proxyAddress) {
+    return new Builder(proxyAddress);
+  }
+
+  /**
+   * The helper class to build an Attributes instance.
+   */
+  public static final class Builder {
+
+    private final InetSocketAddress proxyAddress;
+    @Nullable
+    private String username;
+    @Nullable
+    private String password;
+
+    private Builder(InetSocketAddress proxyAddress) {
+      this.proxyAddress = Preconditions.checkNotNull(proxyAddress, "proxyAddress");
+    }
+
+    public Builder username(String username) {
```

Comment:
`@Nullable` ?

Reply:
fixed

Reply:
The nullable here is still in the wrong location. Ditto below.